solutions/cses-1736.mdx

```diff
@@ -0,0 +1,149 @@
+---
+id: cses-1736
+source: CSES
+title: Polynomial Queries
+author: Albert Ye
+---
+
+## Solution
+**Time Complexity: $\mathcal{O}(n\log n)$**
+
+In this question, we have range updates and range queries, but
+the updates are of a strange format: incrementing the first number
+in interval by $1$, the second by $2$, etc. However, despite the
+strange operations we can still use a segment tree with lazy propagation
+to solve this problem.
+
+The range queries should be similar to in other lazy problems: we finish
+the lazy propagation to all needed nodes and find the range query. However,
+the range updates are different. 
+
+When we update, we want to keep 2 lazy values: one (`lz`) stores the starting value
+and the other (`lz2`) stores the increment. When we propagate, the value of `lz` for
```

Comment:
What do you mean by "starting value" and "increment"?

Reply:
i'm pretty shit at explaining these things, but

starting value = like the basic range sum part where we add 1 to each index
increment = if you're currently adding 3 to a point, then the increment will be 2. basically for each range, the amount you need to increment to get the value that you actually want to add